### ISSUE TITLE
CLISP argv

### ISSUE THREAD

**Issue by bamboospirit** [libre-man/unix-opts#2]
Hi, on CLISP ext:argv is a function, i think unix-opts:argv should be something like this
# +clisp (map 'list (lambda (x) x) (ext:argv)) ; convert (ext:argv) from array to list

or better:
# +clisp ext:*args*

(note that CLISP's ext:*args* doesn't include the binary name. In my scripts i had to manually add "clisp" as an argv parameter to get the same behaviour as with sbcl)

Btw unix-opts is really cool!

**Comment by mrkkrp** [comment]
Thanks for the report, I'll take a look at this soon.

---

According to [this page](http://www.clisp.org/impnotes.html#script-exec), we can get list of command line arguments as `ext:*args*` and name of executable (probably with path) `*load-truename*`. So, maybe we could just:

``` common-lisp
  #+clisp (cons *load-truename* ext:*args*)
```

Current solution is partially taken from `unix-options` from here:

https://github.com/astine/unix-options/blob/master/unix-options.lisp#L279

Obviously their approach is also currently incorrect because `ext:*args*` doesn't contain application name as the first element, I can imagine what kind of bugs this can lead to.

**Comment by mrkkrp** [comment]
Also it seems that output of `ext:argv` is quite different from `ext:*argv*`:

```
[1]> (ext:argv)
#("/usr/lib/clisp-2.49/base/lisp.run" "-B" "/usr/lib/clisp-2.49" "-M"
  "/usr/lib/clisp-2.49/base/lispinit.mem" "-N" "/usr/share/locale")
[2]> ext:*args*
NIL
```

**Comment by bamboospirit** [comment]
Yes #+clisp (cons _load-truename_ ext:_args_) seems okay

**Comment by bamboospirit** [comment]
Or what about using uiop:_command-line-arguments_ ?

**Comment by mrkkrp** [comment]
Well, in this library I prefer to avoid dependencies (and this is easy). `uiop` package should itself use implementation-dependent definition for this thing. We could just steal that.

**Comment by mrkkrp** [comment]
Fixed in 13008e9f.
